components/linking_landing_page/linking_landing_page.tsx

```diff
@@ -0,0 +1,473 @@
+        }
+        this.setState({rememberChecked: !this.state.rememberChecked});
+    }
+
+    setPreference = (pref: string, clearIfNotChecked?: boolean) => {
+        if (!this.state.rememberChecked) {
+            if (clearIfNotChecked) {
+                BrowserStore.clearLandingPreference();
+            }
+            return;
+        }
+
+        switch (pref) {
+        case LandingPreferenceTypes.MATTERMOSTAPP:
+            BrowserStore.setLandingPreferenceToMattermostApp(this.props.siteUrl);
+            break;
+        case LandingPreferenceTypes.BROWSER:
+            BrowserStore.setLandingPreferenceToBrowser(this.props.siteUrl);
+            break;
+        default:
+            break;
+        }
+    }
+
+    openMattermostApp = () => {
+        this.setPreference(LandingPreferenceTypes.MATTERMOSTAPP);
+        this.setState({redirectPage: true});
+        window.location.href = this.state.nativeLocation;
+    }
+
+    openInBrowser = () => {
+        this.setPreference(LandingPreferenceTypes.BROWSER);
+        window.location.href = this.state.location;
+    }
+
+    tryOpen = () => {
+        safeOpenProtocol(this.state.nativeLocation,
+            () => this.setState({protocolUnsupported: true}),
+            () => null,
+            () => this.setState({browserUnsupported: true})
+        );
+    }
+
+    renderSystemDialogMessage = () => {
+        const isMobile = UserAgent.isMobile();
+
+        if (isMobile) {
+            return (
+                <FormattedMessage
+                    id='get_app.systemDialogMessageMobile'
+                    defaultMessage='View in App'
+                />
+            );
+        }
+
+        return (
+            <FormattedMessage
+                id='get_app.systemDialogMessage'
+                defaultMessage='View in Desktop App'
+            />
+        );
+    }
+
+    renderGoNativeAppMessage = () => {
+        const {browserUnsupported, protocolUnsupported} = this.state;
+        const downloadLink = this.getDownloadLink();
+
+        if (protocolUnsupported && downloadLink) {
+            return (
+                <a
+                    href={downloadLink}
+                    className='btn btn-primary btn-lg get-app__download'
+                >
+                    <FormattedMessage
+                        id='get_app.downloadMattermost'
+                        defaultMessage='Download App'
+                    />
+                </a>
+            );
+        } else if (browserUnsupported) {
+            return (
+                <a className='btn btn-primary btn-lg get-app__download disabled'>
+                    <FormattedMessage
+                        id='get_app.browserUnsupported'
+                        defaultMessage='Browser not supported.'
+                    />
+                </a>
+            );
+        }
+
+        return (
+            <a
+                href={this.state.nativeLocation}
+                onMouseDown={() => {
```

Comment:
Why mix `onMouseDown` and `onClick` here?

Reply:
Because `onClick` will not fire if you middle click or Cmd+Click or right click to open in a new tab or window.